// weex-vue-precompiler/test/output/misc/units.js
module.exports = [
  {
    type: 1,
    tag: 'p',
    _origTag: 'text',
    _weexBuiltIn: true,
    plain: false,
    attrs: [
      {
        name: 'weex-type',
        value: '"text"'
      }
    ],
    staticClass: '"txt weex-el weex-text"',
    staticStyle: JSON.stringify({
      'font-size': '32px'
    })
  }, {
    type: 1,
    tag: 'p',
    _origTag: 'text',
    _weexBuiltIn: true,
    plain: false,
    attrs: [
      {
        name: 'weex-type',
        value: '"text"'
      }
    ],
    staticClass: '"txt weex-el weex-text"',
    styleBinding: "{ fontSize: _px2rem('32wx', 75) }"
  }, {
    type: 1,
    tag: 'p',
    _origTag: 'text',
    _weexBuiltIn: true,
    plain: false,
    attrs: [
      {
        name: 'weex-type',
        value: '"text"'
      }
    ],
    staticClass: '"txt wx weex-el weex-text"',
  }, {
    type: 1,
    tag: 'div',
    _origTag: 'div',
    _weexBuiltIn: true,
    plain: false,
    attrs: [
      {
        name: 'weex-type',
        value: '"div"'
      }
    ],
    staticClass: '" weex-ct weex-div"',
    static: false
  }
]
